test/client/fixtures/core/request-barrier-test.js

```diff
@@ -0,0 +1,466 @@
+                    }
+
+                    // NOTE: ignore slow connection on the testing
+                    // farm that leads to unstable tests appearing
+                    delay(requestBarrier.BARRIER_TIMEOUT)
+                        .then(function () {
+                            barrierTimeoutExceeded = true;
+                        });
+
+                    requestBarrier
+                        .wait()
+                        .then(function () {
+                            ok(completeReqCount === reqCount || barrierTimeoutExceeded);
+                            start();
+                        });
+                });
+
+                asyncTest('timeout', function () {
+                    expect(1);
+
+                    var requestBarrier   = new RequestBarrier();
+                    var reqIsCompleted = false;
+
+                    requestBarrier.BARRIER_TIMEOUT = 0;
+
+                    fetch('/xhr-test/8000')
+                        .then(function (response) {
+                            return response.text();
+                        })
+                        .then(function () {
+                            reqIsCompleted = true;
+                        });
+
+                    requestBarrier
+                        .wait()
+                        .then(function () {
+                            ok(!reqIsCompleted);
+                            start();
+                        });
+                });
+
+                asyncTest('should not wait pending requests', function () {
+                    var firstRequestCompleted  = false;
+                    var secondRequestCompleted = false;
+
+                    fetch('/xhr-test/3000')
+                        .then(function (response) {
+                            return response.text();
+                        })
+                        .then(function () {
+                            firstRequestCompleted = true;
+                            start();
+                        });
+
+                    expect(2);
+
+                    var requestBarrier = new RequestBarrier();
+
+                    window.setTimeout(function () {
+                        requestBarrier
+                            .wait()
+                            .then(function () {
+                                ok(!firstRequestCompleted);
+                                ok(secondRequestCompleted);
+                            });
+
+                        fetch('/xhr-test/200')
+                            .then(function (response) {
+                                return response.text();
+                            })
+                            .then(function () {
+                                secondRequestCompleted = true;
+                            });
+                    }, 100);
+                });
+
+                asyncTest('barrier - creating new iframe without src (B236650)', function () {
+                    var $iframe           = null;
```

Comment:
I think this test is redundant (Isn't  it the same like test above https://github.com/DevExpress/testcafe/pull/548/files#diff-5b3b1295f2f653bab8bc18410ccaba81R201)